src/symbol/sprite_atlas.js

```diff
@@ -61,6 +61,8 @@ class SpriteAtlas extends Evented {
 
         if (ArrayBuffer.isView(pixels)) {
             pixels = new Uint32Array(pixels.buffer);
+        } else if (pixels.data) {   //Support ImageData correctly
```

Comment:
You're right, it isn't documented. I was thrown off by the error message at [Line 67](https://github.com/mapbox/mapbox-gl-js/pull/4644/files#diff-69b64c0a443828ff53dfe302a3dffb5bL67) below